back-end/src/base/calls/calls.service.ts

```diff
@@ -35,18 +35,61 @@ export class CallsService {
 	}
 
 	async generateUrl(jwt: string) {
-		return `https://1f25-2a01-cb00-e91-b600-30f3-a390-1412-28a2.eu.ngrok.io/calls/presence/` + jwt;
+		return `https://ce88-2a01-cb00-e91-b600-33-cd57-e77e-fcd8.eu.ngrok.io/calls/presence/` + jwt;
 	}
 
 	async getActualCourse(userId: ObjectId) {
 		const actualCourse = await this.callsRepository.getActualCourse(userId);
 		return actualCourse;
 	}
+	async getStudentList(courseId: CourseIdObject) {
+		const studentIdList = await this.callsRepository.getStudentIdList(courseId.courseId);
+		return this.callsRepository.getStudentList(courseId.courseId, studentIdList);
+	}
+
+	getStudentPresence(courseId: CourseIdObject, studentId: StudentIdObject) {
+		return this.callsRepository.getStudentPresence(courseId.courseId, studentId.studentId);
+	}
+
+	async arrayGenerator(studentAmount: number, courseId: CourseIdObject) {
+		const amountOfHash = this.calculateGroups(studentAmount);
+
+		const finalHash = [];
 
-	getStudentIdList(courseId: CourseIdObject) {
-		return this.callsRepository.getStudentIdList(courseId.courseId);
+		for (let i = 1; i <= amountOfHash['groupsOf4']; i++) {
+			const hash = [];
+			for (let j = 0; j <= 3; j++) {
+				hash.push('');
+			}
+			finalHash.push(hash);
+		}
+
+		for (let i = 1; i <= amountOfHash['groupsOf3']; i++) {
+			const hash = [];
+			for (let j = 0; j <= 2; j++) {
+				hash.push('');
+			}
+			finalHash.push(hash);
+		}
+
+		await this.callsRepository.createGroups(finalHash, courseId.courseId);
+
+		return finalHash;
 	}
-	getStudentList(studentIdList: Array<ObjectId>) {
-		return this.callsRepository.getStudentList(studentIdList);
+
+	calculateGroups(studentAmount: number) {
+		let groupsOf3 = 0;
+		let groupsOf4 = 0;
+
+		while (studentAmount > 0) {
+			if (studentAmount % 3 === 0) {
```

Comment:
This line was modified because you repeated the logic.
I added a comment to be specific tho.